### PR TITLE
Basic Components for Unit Notes Filing 

### PR DESCRIPTION
**Pull request by dimak1**
*Issue #:* 
- bcgov/entity#16508
- bcgov/entity#16517

*Description of changes:*
- Doc Id, Remarks and ContactInformation components
- Basic validation
- Update unit tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1397-3wyv5oky.web.app

File: ppr-ui/src/components/common/ContactInformation.vue [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+      flat
+      rounded
+      class="mt-8 pa-8 pr-6"
+      :class="{ 'border-error-left': validateContactInfo }"
+    >
+      <v-row no-gutters>
+        <v-col cols="12" sm="2">
+          <label
+            class="generic-label"
+            :class="{'error-text': validateContactInfo}"
```
Comment (cameron-eyds) [inline ppr-ui/src/components/common/ContactInformation.vue]: Super minor but spacing should be consistent within the {}. 
I'm good with either 👍

Comment (dimak1) [reply ppr-ui/src/components/common/ContactInformation.vue]: Yea i agree, im fixing the formatting and other small improvements.

File: ppr-ui/src/components/unitNotes/DocumentId.vue [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    return {
+      ...toRefs(localState)
+    }
+  }
+})
+</script>
+
+<style lang="scss" module>
+@import '@/assets/styles/theme.scss';
+</style>
```
Comment (cameron-eyds) [inline ppr-ui/src/components/unitNotes/DocumentId.vue]: Looks great! is it possible to relocate this within the common components and replace the portion in the SubmittingParty.vue? So we just have 1 instance of Document ID.

If you don't want to do within the scope of this ticket, totally understandable but perhaps create a techdebt ticket and try to prioritize with Melissa.

File: ppr-ui/src/components/unitNotes/Remarks.vue [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      maxLength,
+      ...toRefs(localState)
+    }
+  }
+})
+</script>
+
+<style lang="scss" module>
+@import '@/assets/styles/theme.scss';
+</style>
```
Comment (cameron-eyds) [inline ppr-ui/src/components/unitNotes/Remarks.vue]: Same as above, we should have 1 remarks component. Ideally this is common and used in `OtherInformation.vue` as well.

Comment (cameron-eyds) [reply ppr-ui/src/components/unitNotes/Remarks.vue]: If not done here, we should have a techdebt ticket with some visibility.

**Review by cameron-eyds — APPROVED** [review]
Looks great! 
Just a little tech-debt as result of duplicate components for doc id and remarks. 

Your call to handle that here or create a tech-debt ticket to consolidate those components. 
Please create a ticket if you choose to defer it  👍

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1397-3wyv5oky.web.app

File: ppr-ui/src/components/unitNotes/DocumentId.vue (hunk 2 above)
Comment (dimak1) [reply ppr-ui/src/components/unitNotes/DocumentId.vue]: Moved to a common folder and created the tech debt ticket.

File: ppr-ui/src/components/unitNotes/Remarks.vue (hunk 3 above)
Comment (dimak1) [reply ppr-ui/src/components/unitNotes/Remarks.vue]: Create a tech debt ticket.

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1397-3wyv5oky.web.app

**Review by doug-lovett — APPROVED** [review]

File: ppr-ui/src/components/common/DocumentId.vue [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+      displayDocIdError: false,
+      validateDocId: computed(() => props.validate && !localState.isVerifiedDocId),
+      isVerifiedDocId: computed(() => {
+        return localState.isDocumentIdFormValid && localState.isUniqueDocId
+      }),
+      uniqueDocIdError: computed(() => {
+        // Manual error handling for Unique DocId Lookup
+        return localState.displayDocIdError ? ['Must be unique number'] : []
+      }),
+      documentIdRules: computed(() => {
```
Comment (orelbn) [inline ppr-ui/src/components/common/DocumentId.vue]: Do we need to add a required rule?

Update: just looking at the design. I think so.

image

File: ppr-ui/src/components/unitNotes/UnitNoteAdd.vue [hunk 5]
```diff
@@ -1,27 +1,36 @@
...
     <section id="mhr-unit-note-doc-id" class="mt-10">
-      <h2>Document ID</h2>
-      <p class="mt-2">
-        Enter the 8-digit Document ID number.
-      </p>
-      // Add Document ID placeholder //
+      <DocumentId
+        :documentId="unitNoteDocumentId"
+        :setStoreProperty="setDocumentId"
+        :validate="true"
```
Comment (orelbn) [inline ppr-ui/src/components/unitNotes/UnitNoteAdd.vue]: Do we want validate true here? I assume this will be updated during a validation ticket.

**Review by orelbn — APPROVED** [review]
Looks great! just a couple small questions in the comments.

File: ppr-ui/src/components/common/DocumentId.vue (hunk 4 above)
Comment (dimak1) [reply ppr-ui/src/components/common/DocumentId.vue]: Yes, thank you!

File: ppr-ui/src/components/unitNotes/UnitNoteAdd.vue (hunk 5 above)
Comment (dimak1) [reply ppr-ui/src/components/unitNotes/UnitNoteAdd.vue]: Good catch, slipped in by mistake.

**Comment by dimak1** [comment]
@orelbn addressed the comments

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1397-3wyv5oky.web.app

**Pull request merged**